gadgets/spoiler/gadget%3Aspoiler.js

```diff
@@ -16,22 +16,31 @@
   }
   $('.spoiler-js').show();
 
-  $('<a>')
-    .html('보이기')
-    .addClass('spoiler-toggle-button')
-    .click(function (event) {
-      $spoilerTexts.fadeToggle();
-      if ($hasSpoilingHeadings) {
-        $toc.toggle();
-      }
-      $noticeTexts.toggle();
-
-      if ($(this).html() == '숨기기') $(this).html('보이기');
-      else $(this).html('숨기기');
-
-      event.preventDefault();
-    })
-    .appendTo($noticeTexts.parent());
+  var anchor = document.createElement('a');
+  anchor.textContent = '보이기';
+  anchor.classList.add('spoiler-toggle-button');
+  anchor.addEventListener('click', function (event) {
+    $spoilerTexts.fadeToggle();
+    if ($hasSpoilingHeadings) {
+      $toc.toggle();
+    }
+    $noticeTexts.toggle();
+
+    if (this.textContent === '숨기기') {
+      this.textContent = '보이기';
+    } else {
+      this.textContent = '숨기기';
+    }
+
+    event.preventDefault();
+  });
+  $noticeTexts[0].parentElement.append(anchor);
+
+  mw.hook('ve.deactivationComplete').add(function () {
+    $noticeTexts = $('.spoiler-notice-text');
+    $spoilerTexts = $('.spoiler-text');
+    $noticeTexts[0].parentElement.append(anchor);
+  });
```

Comment:
편집 완료 후에는 ToC가 토글되지 않는 문제도 같이 고칠 수 있을까요? 아마 시각편집기에서 아예 새로 만들거나 해서 그런 것 같습니다
```suggestion
    $toc = $('#toc, .toc')
  });
```
이렇게 시험했더니 괜찮았습니다. 그렇게 안 깔끔해 보이긴 한데 어떨까요...?

Reply:
크게 중요한 부분은 아니라서 그냥 바로 머지 원하시면 말씀해주세요~